src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldNotContainUnderscoresTests.cs

```diff
@@ -107,17 +123,71 @@ public class UnderScoreInNameButNotExternallyVisible_
     }
 }
 ",
-            GetCA1707CSharpResultAt(line: 4, column: 18, symbolKind: SymbolKind.NamedType, identifierNames: "OuterType.UnderScoreInName_"));
+            VerifyCS.Diagnostic(IdentifiersShouldNotContainUnderscoresAnalyzer.TypeRule).WithLocation(0).WithArguments("OuterType.UnderScoreInName_"), @"
+public class OuterType
+{
+    public class UnderScoreInName
+    {
+    }
+
+    private class UnderScoreInNameButPrivate_
+    {
+    }
+
+    internal class UnderScoreInNameButInternal_
+    {
+    }
+}
+
+internal class OuterType2
+{
+    public class UnderScoreInNameButNotExternallyVisible_
+    {
+    }
+}
+");
         }
 
         [Fact, WorkItem(1432, "https://github.com/dotnet/roslyn-analyzers/issues/1432")]
         public async Task CA1707_ForFields_CSharp()
         {
-            await VerifyCS.VerifyAnalyzerAsync(@"
+            await VerifyCS.VerifyCodeFixAsync(@"
 public class DoesNotMatter
 {
+        public const int {|#0:ConstField_|} = 5;
+        public static readonly int {|#1:StaticReadOnlyField_|} = 5;
+
+        // No diagnostics for the below
+        private string InstanceField_;
+        private static string StaticField_;
+        public string _field;
+        protected string Another_field;
+}
+
+public enum DoesNotMatterEnum
+{
+    {|#2:_EnumWithUnderscore|},
+    {|#3:_|}
```

Comment:
I thought about that at first, but preferred to keep the warning in this case without a code fix.

But after a second thought, it would probably be okay if we don't report on this case. It's really a very edge case. Almost no one will write something like that 😄 

@mavasani, would you be okay if we don't report on this case?

Reply:
Yes, I don't really care if we report a diagnostic on `_` or not.

Reply:
On the second thought I also think reporting or not are both valid options.